app/handlers/sessions_create.rb

```diff
@@ -57,7 +57,14 @@ def handle
         status = :authentication_taken
       else
         return outputs[:status] = :new_signin_required if user_signin_is_too_old?
-        run(TransferAuthentications, authentication, current_user)
+        auth_transfer = run(TransferAuthentications, authentication, current_user)
+
+        if auth_transfer.outputs[:same_provider]
```

Comment:
I would like to discuss whether (2) can still be switched to a `fatal_error`.  I think it can because with (3) you'll be adding a check before you call `TransferAuthentications`, so the `fatal_error` will never be reached in your situation.